### DISCUSSION TITLE
Run Nextflow inside Docker

### DISCUSSION THREAD

**Discussion by r0f1** [nextflow-io/nextflow#4831] [Q&A]
Hi,

I have the same problem as discussed here: https://github.com/nextflow-io/nextflow/discussions/3782

I would like to run `nextflow run` from inside a webserver that is dockerized. However, the flag `-d / -dockerize` for `nextflow run` is gone now in the latest version of nextflow. [The documentation](https://www.nextflow.io/docs/latest/cli.html) says it is just depricated, but it is gone. 

I found [this article](https://vinodhakumara2681997.medium.com/run-docker-in-docker-42f381fe6b4f) that suggests a solution to the problem by either 1) mounting `/var/run/docker.sock` or 2) using a special image. 

How can I make either 1) or 2) work? Does nextflow allow me that kind of low-level access and/or forwards these additional command line flags?

**Comment by bentsherman** [comment]
I would just study how the dockerize option was implemented in Nextflow and take a similar approach:

https://github.com/nextflow-io/nextflow/blob/973b0310a32d2f87c2f3b4e78847cc348364c26f/nextflow#L226-L237

It is just some Bash code to launch the Nextflow pipeline in a Docker image, with some extra options so that the task containers are launched in the host environment rather than within the Nextflow container. We removed it because it was difficult to maintain in a generic way, but as a custom solution it is feasible.
